app/models/financialDetails/PaymentsResponse.scala

```diff
@@ -45,7 +45,11 @@ case class Payment(reference: Option[String],
                    dueDate: Option[LocalDate],
                    documentDate: LocalDate,
                    transactionId: Option[String],
-                   mainType: Option[String] = None) {
+                   mainType: Option[String] = None,
+                   mainTransaction: Option[String] = None) {
```

Comment:
we might need to introduce a new type here:
class TransactionId(val id: String) extends AnyVal
should be ok to move it into tech debt story as this is really not in scope of this story

Reply:
What is the benefit of this?

Reply:
* this would improve code readability and less chances to make a mistake, 
there are many other reason to do this, 
* we already applying something similar ie: IncomeSourceId and Nino, TaxYear, these basic domain primitives
in general we should avoid using such types as String and Int, there is no meaning in these when you read code, variable name is not something we should rely on as this is not providing type safety